cli_tools/common/utils/logging/logger.go

```diff
@@ -15,50 +15,205 @@
 package logging
 
 import (
-	"fmt"
 	"log"
+	"os"
+	"strings"
+	"sync"
 	"time"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/proto/go/pb"
+	"github.com/golang/protobuf/proto"
 )
 
-// LogEntry encapsulates a single log entry.
-type LogEntry struct {
-	LocalTimestamp time.Time `json:"localTimestamp"`
-	Message        string    `json:"message"`
+// LogWriter is a logger for interactive tools. It supports
+// string messages and structured metrics.
+//
+// Structured metrics are accumulated over the lifespan of the logger.
+//
+// To rebuild the mock, run `go generate ./...`
+//go:generate go run github.com/golang/mock/mockgen -package mocks -source $GOFILE -destination ../../../mocks/mock_logger.go
+type LogWriter interface {
```

Comment:
Logger might be a more natural name, even in the presence of LogWritter interface. It doesn't only write logs, it also collects them. It's also mroe commonly used name for a class of these responsibilities.

Reply:
Resolving, since I think the thread "If this was named just " also addresses this